config/profile.go

```diff
@@ -651,15 +652,19 @@ func (p *Profile) SetRootPath(rootPath string) {
 func (p *Profile) resolveSourcePath(sourceBase string, sourcePaths ...string) []string {
 	var applySourceBase, applyBaseDir pathFix
 
-	// Backup source is NOT relative to the configuration, but to PWD or sourceBase (if not empty)
-	// Applying "sourceBase" if set
-	if sourceBase = strings.TrimSpace(sourceBase); sourceBase != "" {
-		sourceBase = fixPath(sourceBase, expandEnv, expandUserHome)
-		applySourceBase = absolutePrefix(sourceBase)
-	}
-	// Applying a custom PWD eagerly so that own commands (e.g. "show") display correct paths
-	if p.BaseDir != "" {
-		applyBaseDir = absolutePrefix(p.BaseDir)
+	sourceBase = fixPath(sourceBase, expandEnv, expandUserHome)
+	// When "change-working-dir" is set, the source paths are relative to the "source-base"
+	if !p.Backup.ChangeWorkingDir {
```

Comment:
`p.Backup` should be checked for `nil`, e.g. if applying all other comments:

```go
if p.Backup != nil && p.Backup.SourceRelative {
    applySourceBase = nil
    applyBaseDir = nil
}
```

At the moment it wont be nil when this is used but a nil check in this case would be better as the callers of the method can legally use it when the backup section is missing.